crates/orchestrator/src/jobs/mod.rs

```diff
@@ -186,6 +186,33 @@ pub async fn verify_job(id: Uuid) -> Result<()> {
     Ok(())
 }
 
+/// Terminates the job and updates the status of the job in the DB.
+/// Logs error if the job status `Completed` is existing on DL queue.
+pub async fn handle_job_failure(id: Uuid) -> Result<()> {
+    let config = config().await;
+
+    let mut job = get_job(id).await?.clone();
+    let mut metadata = job.metadata.clone();
+
+    if job.status == JobStatus::Completed {
+        log::error!("Invalid state exists on DL queue: {}", job.status.to_string());
+        return Ok(());
```

Comment:
why do we fail silently here ?

Reply:
DL-queue is supposed to handle actual failed cases.
If `JobStatus::Completed` job is pushed to DL-queue multiple times by the queuing agent,
we prefer not stopping the orchestrator rather `failing silently`.